docs/qaoa/binary_paintshop.ipynb

```diff
@@ -0,0 +1,460 @@
+    "from a restricted native gate set predetermined by the\n",
+    "physics of the quantum processor. To execute an arbitrary gate, compilation of the desired gate into available gates is required. For trapped ions, a generic native\n",
+    "gate set consists of a parameterized two-qubit rotation, the Molmer Sorensen gate,\n",
+    "$R_\\mathrm{XX}(\\alpha)=\\mathrm{exp}[-\\mathrm{i}\\alpha \\sigma_\\mathrm{x}^{(i)}\\sigma_\\mathrm{x}^{(j)}/2]$ and a parametrized single qubit rotation:\n",
+    "\n",
+    "$R(\\theta,\\phi)=\\begin{pmatrix}\n",
+    "\\cos{(\\theta/2)} & -\\mathrm{i}\\mathrm{e}^{-\\mathrm{i}\\phi}\\sin{(\\theta/2)} \\\\-\\mathrm{i}\\mathrm{e}^{\\mathrm{i}\\phi}\\sin{(\\theta/2)} &  \\cos{(\\theta/2)}  \n",
+    "\\end{pmatrix}$\n",
+    "\n",
+    "QAOA circuits employ parametrized two body $\\sigma_z$ rotations, $R_\\mathrm{ZZ}(\\gamma)=\\mathrm{exp}[-i\\gamma \\sigma_\\mathrm{z}^{(i)}\\sigma_\\mathrm{z}^{(j)}]$. To circumvent a compilation overhead and optimally leverage the Ion Trap, we inject pairs of Hadamard gates $H H^{\\dagger} = 1$ for every qubit in between the two body $\\sigma_z$ rotations. This means we are able to formulate the phase separator entirely with Molmer Sorensen gates. To support this, the QAOA circuit starts in the state where all qubits are in the groundstate $\\left| 0\\right\\rangle$ instead of the superposition of all computational basis states $\\left| + \\right\\rangle$,"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "uensYoJ1tUpB"
+   },
+   "outputs": [],
+   "source": [
+    "def phase_separator(\n",
+    "    gamma: float, qubit_register: Sequence[cirq.Qid], car_sequence: Sequence[int]\n",
+    ") -> Sequence[cirq.Operation]:\n",
+    "    \"\"\"Yield a sequence of Molmer Sorensen gates to implement a\n",
+    "    phase separator over the ferromagnetic/antiferromagnetic\n",
+    "    interactions between adjacent cars, as defined by spin_glass\n",
+    "    \"\"\"\n",
+    "    for car_pair0, car_pair1, interaction in spin_glass(car_sequence):\n",
+    "        yield cirq.ms(interaction * gamma).on(\n",
+    "            qubit_register[car_pair0], qubit_register[car_pair1]\n",
+    "        )\n",
+    "\n",
+    "\n",
+    "qubit_register = cirq.LineQubit.range(CAR_PAIR_COUNT)\n",
+    "circuit = cirq.Circuit([phase_separator(0.1, qubit_register, car_sequence)])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "5dnVhwjUk3GT"
+   },
+   "source": [
+    "Because we replaced the two body $\\sigma_z$ rotations with Molmer Sorensen gates we also have to adjust the mixer slightly to account for the injected Hadamard gates."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "WSo8TGlOwgko"
+   },
+   "outputs": [],
+   "source": [
+    "def mixer(beta, qubit_register):\n",
+    "    yield cirq.ry(np.pi / 2).on_each(qubit_register), cirq.rx(beta - np.pi).on_each(\n",
```

Comment:
Consider splitting this into two lines for clarity:

```
yield cirq.ry(np.pi / 2).on_each(qubit_register)
yield cirq.rx(beta - np.pi).on_each(qubit_register)
```

(Also double-check that doesn't break anything.)